ui/src/components/reuseable/editDialog.tsx

```diff
@@ -1,90 +1,226 @@
 
 export interface EditProps {
-  filename: string;
   name: string;
+  id: number;
   emoji: string;
   color: string;
-  tags: string[];
-  folders: string[];
+  tags: Tag[];
+  folders: Folder[];
 }
 
 export interface EditDialogProps
-  extends Partial<EditProps>,
-    React.ComponentPropsWithoutRef<typeof Dialog.Root> {
+  extends React.ComponentPropsWithoutRef<typeof Dialog.Root> {
   uploadFirst?: boolean;
   className?: string;
-  onOpenChange?: (open: boolean) => void;
+  open: boolean;
+  onOpenChange: (open: boolean) => void;
+  sound?: EditProps;
 }
 
+const defaultSound: EditProps = {
+  name: "",
+  id: -1,
+  emoji: "🎉",
+  color: "#bb27ff",
+  tags: [],
+  folders: [],
+};
+
 const EditDialog = ({
   uploadFirst = false,
-  filename = "",
-  name = "",
-  emoji = "🎉",
-  color = "#bb27ff",
-  tags = [],
-  folders = [],
+  sound = defaultSound,
+  open,
   onOpenChange,
   className = "",
   children,
   ...props
 }: EditDialogProps) => {
   const classes = `editDialog ${className}`.trim();
   const [uploading, setUploading] = useState(uploadFirst);
-  const [editingFilename, setEditingFilename] = useState(filename);
-  const [editingName, setEditingName] = useState(name);
-  const [editingEmoji, setEditingEmoji] = useState(emoji);
-  const [editingColor, setEditingColor] = useState(color);
-  const [editingTags, setEditingTags] = useState(tags);
-  const [editingFolders, setEditingFolders] = useState(folders);
+  const [editingFile, setEditingFile] = useState<File | null>(null);
+  const [editingName, setEditingName] = useState(sound.name);
+  const [editingEmoji, setEditingEmoji] = useState(sound.emoji);
+  const [editingColor, setEditingColor] = useState(sound.color);
+  const [editingTags, setEditingTags] = useState<Tag[]>(sound.tags);
+  const [editingFolders, setEditingFolders] = useState<Folder[]>(sound.folders);
+  const [isSaving, setIsSaving] = useState(false);
+
+  type CreateEditSoundProps = {
+    name: string;
+    color: string;
+    emoji: string;
+    tags: number[];
+    folders: string[];
+    audio_file?: File;
+  };
+
+  const editSoundMutation = useMutation({
+    mutationFn: async (editedSound: CreateEditSoundProps) => {
+      const formData = new FormData();
+      formData.append("sound[name]", editedSound.name);
+      formData.append("sound[color]", editedSound.color);
+      formData.append("sound[emoji]", editedSound.emoji);
+      editedSound.tags.forEach((tagId: number) => {
+        formData.append("sound[tag_ids][]", tagId.toString());
+      });
+      editedSound.folders.forEach((folderSlug: string) => {
+        formData.append("sound[folder_slugs][]", folderSlug);
+      });
+      if (editedSound.audio_file) {
+        formData.append("sound[audio_file]", editedSound.audio_file); // key matches model attribute
+      }
+
+      const res = await fetch(
+        `${import.meta.env.VITE_API_HOST}:${
+          import.meta.env.VITE_API_PORT
+        }/sounds/${sound.id}`,
```

Comment:
a helper `const` for `${import.meta.env.VITE_API_HOST}:${import.meta.env.VITE_API_PORT}` as a base api URL could make sense. you could also set up a fetch helper with that base URL included